src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,34 +33,102 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] splitTrimmedAddress = trimmedAddress.split(ADDRESS_DELIMITER);
+        initAddressSubclasses(splitTrimmedAddress);
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Initialises various subclasses of an address (block, street, unit, postal code).
+     * 
+     * @param splitTrimmedAddress trimmed address split into the various subclasses
+     */
+	private void initAddressSubclasses(String[] splitTrimmedAddress) {
+		this.block = new AddressBlock(splitTrimmedAddress[0]);
+		this.street = new AddressStreet(splitTrimmedAddress[1]);
+		this.unit = new AddressUnit(splitTrimmedAddress[2]);
+		this.postalCode = new AddressPostalCode(splitTrimmedAddress[3]);
+	}
+
+    /**
+     * Returns true if a given string is a valid person address.
+     * Allows for postal codes of any length to facilitate overseas addresses.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    	return hasCorrectNumOfParams(test);
+    }
+    
+    /**
+     * Checks if address specified in user input has the correct number of parameters.
+     */
+    public static boolean hasCorrectNumOfParams(String test) {
+    	String[] splitTest = test.split(ADDRESS_DELIMITER);
+    	return splitTest.length == 4;
     }
 
-    @Override
-    public String toString() {
-        return value;
+    
+    public String toString(AddressBlock block, AddressStreet street,
+            AddressUnit unit, AddressPostalCode postalCode) {
+    	StringJoiner stringJoiner = new StringJoiner(ADDRESS_DELIMITER, "", "");
+		stringJoiner.add(block.getBlock()).add(street.getStreet()).add(unit.getUnit()).add(postalCode.getPostalCode());
```

Comment:
A bit hard to read the code here, May be better to separate them into multiple lines.